bindings/node/index.d.ts

```diff
@@ -0,0 +1,225 @@
+import type { Binary } from 'bson';
+
+export type ClientEncryptionDataKeyProvider = 'aws' | 'local';
+
+/**
+ * An error indicating that something went wrong specifically with MongoDB Client Encryption
+ */
+export class MongoCryptError extends Error {
+}
+
+export interface ClientEncryptionCreateDataKeyCallback {
+  /**
+   * @param error If present, indicates an error that occurred in the creation of the data key
+   * @param dataKeyId If present, returns the id of the created data key
+   */
+  (error?: Error, dataKeyId?: Binary): void;
+}
+
+export interface ClientEncryptionEncryptCallback {
+  /**
+   * @param error If present, indicates an error that occurred in the process of encryption
+   * @param result If present, is the encrypted result
+   */
+  (error?: Error, result?: Binary): void;
+}
+
+export interface ClientEncryptionDecryptCallback {
+  /**
+   * @param error If present, indicates an error that occurred in the process of decryption
+   * @param result If present, is the decrypted result
+   */
+  (error?: Error, result?: any): void;
+}
+
+/**
+ * Configuration options that are used by specific KMS providers during key generation, encryption, and decryption.
+ */
+export interface KMSProviders {
+  /**
+   * Configuration options for using 'aws' as your KMS provider
+   */
+  aws?: {
+    /**
+     * The access key used for the AWS KMS provider
+     */
+    accessKeyId?: string | undefined;
+
+    /**
+     * The secret access key used for the AWS KMS provider
+     */
+    secretAccessKey?: string | undefined;
+  };
+
+  /**
+   * Configuration options for using 'local' as your KMS provider
+   */
+  local?: {
+    /**
+     * The master key used to encrypt/decrypt data keys. A 96-byte long Buffer.
+     */
+    key?: Buffer | undefined;
+  };
+}
+
+/**
+ * Optional settings to provide when creating a new `ClientEncryption` instance.
+ */
+export interface ClientEncryptionOptions {
+  /**
+   * The namespace of the key vault, used to store encryption keys
+   */
+  keyVaultNamespace?: string | undefined;
+
+  /**
+   * Options for specific KMS providers to use
+   */
+  kmsProviders?: KMSProviders | undefined;
+}
+
+/**
+ * Options to provide when creating a new data key.
+ */
+export interface ClientEncryptionCreateDataKeyProviderOptions {
+  /**
+   * Idenfities a new KMS-specific key used to encrypt the new data key. If the kmsProvider is "aws" it is required.
+   */
+  masterKey?: {
```

Comment:
See this discussion over types/future type for masterKey: https://docs.google.com/document/d/1v0UUDvK4Ch8WM03XAFw1Ci-hem84vrgIw07RWCwmyJ4/edit?disco=AAAALAO7L-U I'm not sure if this has happened yet

Reply:
Just reread that and it's possible he mean to comment about masterKey and not kms....really not sure

Reply:
@aherlihy Yes, I put the comment on the wrong part of the type signature.